### PR TITLE
301 run dtaas with localhost domain name

### PR DESCRIPTION
**Pull request by nichlaes**
fixes issue #301 and adds local script

### CODE REVIEW

File: deploy/config/client/env.local.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+window.env = {
```
Comment (prasadtalasila) [inline deploy/config/client/env.local.js:1]: Please [install the pre-commit hooks](https://github.com/INTO-CPS-Association/DTaaS/blob/feature/distributed-demo/docs/developer/index.md) added recently. The pre-commit hook will convert the double quotes to single quotes.

File: deploy/config/gateway/fileConfig.local.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        - http
+      rule: "Host(`localhost`)"
+      middlewares:
+        - basic-auth
+      service: dtaas
+
+    user1:
+      entryPoints:
+        - http
+      rule: "Host(`localhost`) && PathPrefix(`/user1`)"
```
Comment (prasadtalasila) [inline deploy/config/gateway/fileConfig.local.yml:14]: one user is sufficient

File: deploy/config/gateway/fileConfig.local.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        - basic-auth
+      service: user2
+
+    libms:
+      entryPoints:
+        - http
+      rule: "Host(localhost`) && PathPrefix(`/lib`)"
+      service: libms
+
+  # Middleware: Basic authentication
```
Comment (prasadtalasila) [inline deploy/config/gateway/fileConfig.local.yml:33]: This middleware config can be removed for localhost setup

File: deploy/config/gateway/fileConfig.local.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      loadBalancer:
+        servers:
+          - url: "http://localhost:4000"
+
+    user1:
+      loadBalancer:
+        servers:
+          - url: "http://localhost:8090"
+
+    user2:
```
Comment (prasadtalasila) [inline deploy/config/gateway/fileConfig.local.yml:51]: This user2 service will go away too.

File: deploy/local.sh [hunk 5]
```diff
@@ -0,0 +1 @@
+source ./deploy/single-script-install.sh --env local
```
Comment (prasadtalasila) [inline deploy/local.sh:1]: please remove this file. A dedicated documentation page can describe this command with other details of this localhost setup. The documentation can go to _docs/admin/localhost.md_.

**Review by prasadtalasila — COMMENTED** [review]
please fix the codeclimate comments and add documentation. Thanks.

**Comment by nichlaes** [comment]
@prasadtalasila  everything should be resolved now

File: docs/admin/localhost.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+the DTaaS software with a user.
+You can use it to check a test installation of DTaaS software.
+
+## Pre-requisites
+
+### 1. Gitlab OAuth application
+
+The DTaaS react website requires Gitlab OAuth provider.
+If you need more help with this step, please see
+the [Authentication page](client/auth.md).
```
Comment (prasadtalasila) [inline docs/admin/localhost.md:22]: It is sufficient to have [user-owned oauth application](https://docs.gitlab.com/ee/integration/oauth_provider.html#create-a-user-owned-application) for localhost and trial installations. Pæease mention this point in information box both here and on trial installation page.

Please also create a section at the 3nd of _auth.md_  mentioning about the use of user-owned oauth application.

File: docs/admin/trial.md [hunk 7]
```diff
@@ -49,11 +49,12 @@ Remember to create gitlab accounts for `user1` and `user2`.
...
     which services should be restarted. Just click **OK** to all of those.
 
-Run the following scripts.
+Run the following scripts. To setup the installation to use trial,
+the following argument is added to the script `--env trial`.
 
 ```bash
 wget https://raw.githubusercontent.com/INTO-CPS-Association/DTaaS/feature/distributed-demo/deploy/single-script-install.sh
-bash single-script-install.sh
+bash single-script-install.sh --env trial
```
Comment (prasadtalasila) [inline docs/admin/trial.md:57]: If we are using this flag for localhost installation scenario, perhaps we should call this "--localhost".

File: mkdocs.yml [hunk 8]
```diff
@@ -22,6 +22,7 @@ nav:
       - Host Install:
         - Trial: admin/trial.md
         - Production: admin/host.md
+        - Localhost: admin/localhost.md
```
Comment (prasadtalasila) [inline mkdocs.yml:25]: Please sdd this before trial.md page

File: docs/admin/trial.md (hunk 7 above)
Comment (nichlaes) [reply docs/admin/trial.md:57]: The above scenario is for trial

The localhost scenario is `--env local`

What would the trial scenario be?

File: docs/admin/localhost.md (hunk 6 above)
Comment (nichlaes) [reply docs/admin/localhost.md:22]: @prasadtalasila  I don't understand the second part about the auth.md for the use of user-owned oauth

Also what the 3nd section of auth.md is

**Comment by prasadtalasila** [comment]
Continued in PR #380

**Pull request closed without merging**